apps/wallet/src/features/Send/ConfirmByPasscode/index.tsx

```diff
@@ -52,6 +52,10 @@ const PasscodeInput: FC<Props> = ({ navigator, item, activatedId }) => {
 
 			try {
 				const res = await createAndSend(payload, passcode);
+				if (res.error) {
+					throw res.error;
+				}
+
```

Comment:
does not throw error here, createAndSend should throw Error inside it (for web/ext, it will throw error from messaging checking, double check the implementation on mobile)

Reply:
```ts
			try {
				const res = await createAndSend(payload, passcode);

				txActions.update({ time: new Date() });
				txActions.update({ status: res.responseCode });

				if (res.responseCode == ResponseCode.WRONG_PASSCODE) {
					showError({ errorText: 'Passcode is NOT matched' });
					setError('Wrong passcode');
				} else if (res.responseCode == ResponseCode.SUCCESS) {
					const signature =
						res.signatureString || res.signedTransaction?.digest || res.hash;
					txActions.update({ signatureString: signature });
					navigator.slideNext();
				} else if (res.responseCode == ResponseCode.ERROR) {
					// In the case the responseCode is Error and it does not trigger that `catch` block,
					// What should I do?
					// May I throw error here?
				}
			} catch (error) {
				showError({ errorText: (error as Error).message });
			}

			setPasscode('');
		} else if (passcode.length > 0 && error) {
			setError('');
		}
```

Reply:
If not handle the `res.responseCode == ResponseCode.ERROR`, the user will be stuck at the `Passcode` screen

Reply:
for the web, we handled throwing errors with Error Response from the messaging module, you need to throw the error on the mobile (kernel handling) to make things sync